### ISSUE TITLE
Reg\Operation & Facility - Internal user can issue a BCGHGID

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#2152]
#### Dependencies

- Blocked by
- Blocking

#### Description:

BCGHGIDs are pre-populated from the SWRS import, but can be issued by an internal user if no SWRS data existed for it

An External User cannot edit their BORO ID or BCGHG ID. They can only see it in view-only mode.

#### Acceptance Criteria:

Given I am an Internal User,
When I am on the Operation Information Page
Then I can see 2 fields BORO ID and BCGHGID
[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18140-836&t=mTkLtgWBHHXY0SD9-4)

If no BCGHGID has been issued,
then there is a button that says "Issue BCGHGID".

Given that I clicked the button, the following things appear
the BCGHGID
"BCGHGID issued"

Given I am an External User
When I am on my Operation Information Page
Then I see a read-only, and non-editable field for BCGHGID
[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18025-1336&t=mTkLtgWBHHXY0SD9-4)

Given I am an External User
When I am on my LFO's Facility Information Page
Then I see a read-only, and non-editable field for BCGHGID
[Figma](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18025-958&t=mTkLtgWBHHXY0SD9-4)

#### Additional context:

- SFOs receive one BCGHGID for their operation (not on their "facility")
- LFOs receive one BCGHGID for their operation as a whole, and a different BCGHGID for each of the facilities under that LFO (the issuance of each BCGHGID for facilities is done separately. Facilities: [Figma wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18140-1025&m=dev&t=i9soZgPn3FrXjhsF-1). Operation: [Figma wireframe](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=18140-836&m=dev&t=i9soZgPn3FrXjhsF-1) 
- EIOs receive one BCGHGID for their operation (not on their "facility")
- Small Aggregate facilities get a BCGHGID that identifies the entire aggregate of facilities
- The logic for BCGHGID is Divide all facilitie by NAICS code. Starting digit is ‘2’ for an SFO and ‘1’ for LFOs + NAICS code + sequential list of 4 digits (arithmetic progression with the fixed digit being ‘0’)
- Undo functionality is dealt with in ticket #2396 

#### Development Checklist:

- [x] create new model for BCGHGID
- [x] change existing CharField in `Operation` and `Facility` records to`OneToOneField` reference to the assigned BCGHGID (if one exists) - this is consistent with what we've done for assigning BORO IDs
- [x] create new API endpoint for generating and assigning BCGHG ID
- [x] add constraint to DB so that BCGHG IDs are always unique
- [x] functions to generate BCGHG ID
- [x] add pytests for BCGHG ID generation and assignment to facilities and operations
- [x] BCGHG should appear on administration forms (not in the registration workflow)
- [x] BCGHG widget
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

**Comment by patrickisaac** [comment]
Refinement Notes:
- Dev checklist to be added by Andrea

**Comment by BCerki** [comment]
@zoeyli-46 , is there any messaging you want to show for the various cases like we did for BORO ID?

**Comment by vesselak** [comment]
tested in https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/onboarding I can see the 2 fields BORO ID and BCGHGID
I can see BCGHG ID issued message after completing the action
